tests/extmod/ssl_keycert.py.exp

```diff
@@ -1,3 +1,4 @@
 ValueError('invalid key',)
 TypeError("can't convert 'NoneType' object to str implicitly",)
 ValueError('invalid cert',)
+ValueError('invalid cert',)
```

Comment:
This test result really confuses me! Most of the ports (all but esp32) don't have MBEDTLS_PARSE_PEM enabled by default, so you'd expect `mbedtls_parse_key()` to fail and the exception to come back `invalid key` not `invalid cert`. Wouldn't you?

From a casual look in the code I can't see why it's accepting PEM here (i.e. on the unix port default config). Do you know?

Reply:
This is not a fix for PEM, this data passed here is a DER formatted `PKCS#8` key.

See #14371 , this tests DER data in the `PKCS#8` format which without the fix in this or your PR fails because the +1 in the key length passes additional data which mbedtls now checks against (since v3.5). 
But with this it should pass on all ports (I only tested unix and stm32).

Reply:
Ah yes, my mistake that test data is very clearly not in PEM format! Got it, looks good.